Samples/LightCore.Integration.AspNetCore.Sample/Startup.cs

```diff
@@ -26,40 +18,19 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
-
-            services.AddLogging();
-            services.Configure<Person>(o =>
-            {
-                o.FirstName = "";
-                o.LastName = "";
-            });
-            //services.AddMvc();
+            services.AddApplicationInsightsTelemetry();
+            
+            services.AddMvc();
 
 
         }
 
         public void ConfigureContainer(ContainerBuilder builder)
         {
-            // {System.Collections.Generic.IEnumerable`1[Microsoft.Extensions.Options.IPostConfigureOptions`1[Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions]] postConfigures}
-            builder.RegisterFactory<IEnumerable<IPostConfigureOptions<KestrelServerOptions>>>(c => { return new List<IPostConfigureOptions<KestrelServerOptions>>(); });
-            builder.RegisterFactory< IEnumerable<IConfigureOptions<LibuvTransportOptions>>>(x => { return new List<IConfigureOptions<LibuvTransportOptions>>(); });
-            builder.RegisterFactory< IEnumerable<IPostConfigureOptions<LibuvTransportOptions>>>(x => { return new List<IPostConfigureOptions<LibuvTransportOptions>>(); });
-            builder.RegisterFactory< IEnumerable<IConfigureOptions<ConsoleLoggerOptions>>>(x => { return new List<IConfigureOptions<ConsoleLoggerOptions>>(); });
-            builder.RegisterFactory< IEnumerable<IPostConfigureOptions<ConsoleLoggerOptions>>>(x => { return new List<IPostConfigureOptions<ConsoleLoggerOptions>>(); });
-            builder.RegisterFactory< IEnumerable<IOptionsChangeTokenSource<ConsoleLoggerOptions>>>(x => { return new List<IOptionsChangeTokenSource<ConsoleLoggerOptions>>(); });
-
-            builder.RegisterFactory<IEnumerable<IConfigureOptions<ApplicationInsightsServiceOptions>>>(x => { return new List<IConfigureOptions<ApplicationInsightsServiceOptions>>(); });
-            builder.RegisterFactory<IEnumerable<IPostConfigureOptions<ApplicationInsightsServiceOptions>>>(x => { return new List<IPostConfigureOptions<ApplicationInsightsServiceOptions>>(); });
-
-            // {System.Collections.Generic.IEnumerable`1[Microsoft.ApplicationInsights.AspNetCore.ITelemetryProcessorFactory] telemetryProcessorFactories}
-            builder.RegisterFactory<IEnumerable<ITelemetryProcessorFactory>>(x => { return new List<ITelemetryProcessorFactory>(); });
-
-            //{ Void.ctor(
-            //    System.Collections.Generic.IEnumerable`1[Microsoft.Extensions.Options.IConfigureOptions`1[Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.LibuvTransportOptions]], 
-            //    System.Collections.Generic.IEnumerable`1[Microsoft.Extensions.Options.IPostConfigureOptions`1[Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.LibuvTransportOptions]])}
-
-
-            //builder.Register<IService, MyService>();
+            // small hack because this cannot resolved yet. Lightcore wont resolve unknown types
```

Comment:
How to fix this?

Reply:
Tell MS not to add telemetry to ASP.NET Core by default ;-) Just kidding.
I'll check this. Maybe I will add this to the ASP.NET Core LightCore registration as a short term fix.
Just to get the preview 1 released...

Reply:
Allright, but please hold it in mind for later fixing it properly.